src/flexible_type/flexible_type_converter.hpp

```diff
@@ -10,657 +10,899 @@
+    if(src.get_type() == flex_type_enum::FLOAT) {
+      dest = static_cast<Float>(src.get<flex_float>());
+    } else if(src.get_type() == flex_type_enum::INTEGER) {
+      dest = static_cast<Float>(src.get<flex_int>());
     } else {
-      std::string errormsg = 
-          std::string("Expecting a list or array, but we got a ") 
-          + flex_type_enum_to_name(val.get_type());
-      throw(errormsg);
+      throw_type_conversion_error(src, "numeric");
     }
   }
-  flexible_type set(const flex_list& val) {
-    return flexible_type(val);
+
+  template <typename Float>
+  static void set(Float& dest, const flexible_type& src) {
+    dest = flex_float(src);
   }
 };
 
+////////////////////////////////////////////////////////////////////////////////
 
-/**
- * Case 2: Cover's flexible_type itself
+/** Integer values.
  */
-template <>
-struct flexible_type_converter<flexible_type, void> {
-  static constexpr bool value = true;
-  flexible_type get(const flexible_type& val) {
-    return val;
+template <> struct ft_converter<CVTR__INTEGER> {
+
+  template <typename T> static constexpr bool matches() {
+    return std::is_integral<T>::value || std::is_same<T, bool>::value;
   }
-  flexible_type set(const flexible_type& val) {
-    return val;
+
+  template <typename Integer>
+  static void get(Integer& dest, const flexible_type& src) {
+    if(src.get_type() == flex_type_enum::FLOAT) {
+      flex_float v = src.get<flex_float>();
+      if(static_cast<Integer>(v) != v) {
+        throw_type_conversion_error(src, "integer / convertable float");
+      }
+      dest = static_cast<Integer>(v);
+    } else if(src.get_type() == flex_type_enum::INTEGER) {
+      dest = static_cast<Integer>(src.get<flex_int>());
+    } else {
+      throw_type_conversion_error(src, "integer");
+    }
+  }
+
+  template <typename Integer>
+  static void set(Integer& dest, const flexible_type& src) {
+    dest = Integer(flex_int(src));
   }
 };
 
+////////////////////////////////////////////////////////////////////////////////
+//
+//  Vectors
+//
+////////////////////////////////////////////////////////////////////////////////
 
-/**
- * Case 3: All numeric types.
+/** Any sequence containers (vectors, lists, deques, etc.) of floating
+ *  point values.
  */
-template <typename T>
-struct flexible_type_converter<T, 
-    typename std::enable_if<std::is_arithmetic<T>::value>::type> {
-  static constexpr bool value = true;
-  T get(const flexible_type& val) {
-    if (val.get_type() != flex_type_enum::INTEGER && 
-        val.get_type() != flex_type_enum::FLOAT) {
-      std::string errormsg = 
-          std::string("Expecting a numeric type, But we got a ") 
-          + flex_type_enum_to_name(val.get_type());
-      throw(errormsg);
+template <> struct ft_converter<CVTR__FLEX_VEC_CONVERTIBLE_SEQUENCE> {
+
+  template <typename V> static constexpr bool matches() {
+    return (is_sequence_container<V>::value
```

Comment:
Previous version of this captured ```is_arithmetic<typename first_nested_type<V>>``` here to provide a fast-ish path for ```std::vector<int>``` etc. The current path for this will go through CVTR__FLEX_LIST_CONVERTIBLE_SEQUENCE which the current implementation will incur a conversion to flexible_type for each element.

Reply:
Good point.  I can add that.